yt/frontends/parthenon/data_structures.py

```diff
@@ -0,0 +1,342 @@
+        self.domain_width = self.domain_right_edge - self.domain_left_edge
+        self.domain_dimensions = self._handle["Info"].attrs["RootGridSize"]
+
+        self._field_map = {}
+        k = 0
+
+        dnames = self._handle["Info"].attrs["OutputDatasetNames"]
+        num_components = self._handle["Info"].attrs["NumComponents"]
+
+        if "OutputFormatVersion" in self._handle["Info"].attrs.keys():
+            self.output_format_version = self._handle["Info"].attrs[
+                "OutputFormatVersion"
+            ]
+        else:
+            self.output_format_version = -1
+
+        # Use duck typing to check if num_components is a single int or a list
+        # h5py will return different types if there is a single variable or if there are more
+        try:
+            iter(num_components)
+        except TypeError:
+            dnames = (dnames,)
+            num_components = (num_components,)
+
+        component_name_offset = 0
+        for dname, num_component in zip(dnames, num_components):
+            for j in range(num_component):
+                fname = self._handle["Info"].attrs["ComponentNames"][
+                    j + component_name_offset
+                ]
+                self._field_map[fname] = (dname, j)
+                k += 1
+            component_name_offset = int(component_name_offset + num_component)
+
+        self.refine_by = 2
+        dimensionality = 3
+        if self.domain_dimensions[2] == 1:
+            dimensionality = 2
+        if self.domain_dimensions[1] == 1:
+            dimensionality = 1
+        self.dimensionality = dimensionality
+        self.current_time = self._handle["Info"].attrs["Time"]
+        self.num_ghost_zones = 0
+        self.field_ordering = "fortran"
+        self.boundary_conditions = [1] * 6
+
+        if "periodicity" in self.specified_parameters:
+            self._periodicity = tuple(self.specified_parameters["periodicity"])
+        else:
+            boundary_conditions = self._handle["Info"].attrs["BoundaryConditions"]
+
+            inner_bcs = boundary_conditions[::2]
+            # outer_bcs = boundary_conditions[1::2]
+            ##Check self consistency
+            # for inner_bc,outer_bc in zip(inner_bcs,outer_bcs):
+            #    if( (inner_bc == "periodicity" or outer_bc == "periodic") and inner_bc != outer_bc ):
+            #        raise Exception("Inconsistent periodicity in boundary conditions")
+
+            self._periodicity = tuple(bc == "periodic" for bc in inner_bcs)
+
+        gamma_attrs = {
+            key: val
+            for key, val in self._handle["Params"].attrs.items()
+            if key.endswith("AdiabaticIndex")
+        }
+        if "gamma" in self.specified_parameters:
+            self.gamma = float(self.specified_parameters["gamma"])
+        elif len(gamma_attrs) >= 1:
+            self.gamma = next(iter(gamma_attrs.values()))
+        else:
+            self.gamma = 5.0 / 3.0
```

Comment:
It's due to the actual parameter being `Hydro/AdiabaticIndex` in AthenaPK which will be different for the other Parthenon codes. 

If this logic is too opaque, we should instead look for `Hydro/AdiabaticIndex` for AthenaPK and implement the adiabatic index as needed for the other codes.

Reply:
As it's supposed to work right now, an adiabatic index passed in via `gamma=` will take priority over an adiabatic index defined in the parameters of the Parthenon simulation, then `5./3.` if nothing else is found.

Reply:
The `He_mass_fraction` piece isn't working since at some point we stopped outputing `He_mass_fraction` and started including `mbar_over_kb` instead. We could back out `He_mass_fraction` from `mbar_over_kb` instead.

Reply:
I'm trying to fix reading in `He_mass_fraction` from `mbar_over_kb` but I'm getting nonsensical results. I'll have to look at it later with fresh eyes. I'll push what I'm working with in case somone can see what I'm doing wrong.